src/pybind/mgr/snap_schedule/fs/schedule.py

```diff
@@ -239,9 +239,9 @@ def list_schedules(cls,
             else:
                 c = db.execute(cls.PROTO_GET_SCHEDULES + ' path = ?',
                                (f'{path}',))
-        return [cls._from_db_row(row, fs) for row in c.fetchall()]
+            return [cls._from_db_row(row, fs) for row in c.fetchall()]
 
-    INSERT_SCHEDULE = '''INSERT INTO
+    INSERT_SCHEDULE = '''INSERT OR IGNORE INTO
```

Comment:
Why do we need this change?

Reply:
As far as I could find docs on, without that additional clause, the python code dumps a `Traceback` while attempting to add a non unique row. The Python exception is expected as per design.
However, the `Traceback` is dumped even when there's an `except` clause to catch this every exception. This was the weird part.

I'm not sure if the traceback an artifact of Python 3.9 on my laptop when I started testing this piece of code. Teuthology systems run Python 3.6.
(I'm now running Python 3.10 after upgrade to Fedora 35)

I'm also not sure how a SQL DML statement can disregard the actual code in place and force a Traceback to be emitted by the Python run-time.

Reply:
AFAICT, the code already considers the possibility of an exception thrown for an duplicate row. See the `except sqlite3.IntegrityError` below.

Reply:
The changes seemed to pass teuthology tests. So I presumed all's well.
I'll run a few tests independent of cephfs and see how the `OR IGNORE` clause interacts with `except sqlite3.IntegrityError` and get back here.

Reply:
[The original code without `IGNORE` clause works on teuthology](https://pulpito.ceph.com/mchangir-2021-11-24_04:15:48-fs-wip-mchangir-mgr-snap_schedule-fix-db-connection-concurrent-usage-distro-default-smithi/)

Oddly, the code works as expected by throwing and catching the `sqlite3.IntegrityError` exception in a standalone test app as well. But not in a `vstart_runner.py` environment on my laptop. This is really weird.

If somebody is up to it, please try on a Python 3.10 and `vstart_runner.py` environment.
I've attempted without success on a CentOS 8.3 virtual machine as well. Here, `vstart_runner.py` bails out when attempting to mount `/sys/fs/fuse/connections`.

I've now revered back to the original.